app/models/merchant.rb

```diff
@@ -2,4 +2,6 @@ class Merchant < ApplicationRecord
   validates_presence_of :name
   has_many :items
   has_many :invoices
+  has_many :invoice_items, through: :invoices
```

Comment:
New relationships formed to make the AR query easier for returning merchants by most revenue